src/iomei.cpp

```diff
@@ -816,21 +820,34 @@ bool MEIOutput::WriteObject(Object *object)
 
 bool MEIOutput::WriteObjectEnd(Object *object)
 {
-    if (m_scoreBasedMEI && object->IsBoundaryElement()) {
-        return true;
-    }
     // Object representing an attribute have no node to pop
-    else if (object->IsAttribute()) {
-        return true;
-    }
-    else if (m_scoreBasedMEI && (object->Is(SYSTEM))) {
-        return true;
-    }
-    else if (m_scoreBasedMEI && (object->Is(PAGE))) {
+    if (object->IsAttribute()) {
         return true;
     }
-    else if (m_scoreBasedMEI && (object->Is(PAGES))) {
-        return true;
+
+    if (m_scoreBasedMEI) {
+        if (object->Is({ PAGE, PAGES, SYSTEM })) {
+            return true;
+        }
+        if (object->Is({ MDIV, SCORE })) {
+            if (m_page >= 0) {
+                return true;
+            }
```

Comment:
Do you know when this is not true?

Reply:
This is only true when exporting a single page. The page number is zero based, so `m_page=0` means exporting page 1 and so on. When exporting all pages, then `m_page=-1` and the condition is false.